### PR TITLE
ipfire-location-database: init at 2023-06-03

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes
This is a free geoip database which can replace the proprietary GeoLite2 database used by clash and v2ray.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@NickCao 

This database is converted from the one converted by Tor.

### CODE REVIEW

File: pkgs/data/misc/ipfire-geoip/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  inherit (tor.geoip) version;
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ python3 mmdbctl ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    python3 ${./csv-convertor.py} ${tor.geoip}/share/tor/geoip geoip.csv
```
Comment (NickCao) [inline pkgs/data/misc/ipfire-geoip/default.nix:19]: Or shall we directly download the database from https://location.ipfire.org/databases/1/archive/

Comment (linsui) [reply pkgs/data/misc/ipfire-geoip/default.nix:19]: We can download it from https://github.com/torproject/tor/tree/main/src/config. If we download the original database we need to package https://github.com/torproject/tor/tree/main/scripts/maint/geoip and location to convert it from .db to .txt to the Tor csv.

**Comment by NickCao** [comment]
So this should be no longer necessary with https://github.com/NixOS/nixpkgs/pull/235774 and https://github.com/NixOS/nixpkgs/pull/235800 merged?

**Comment by linsui** [comment]
Yes, I thought so.

**Pull request closed without merging**